--- device/pcie/silicon_tlb_window.cpp
+++ device/pcie/silicon_tlb_window.cpp
@@ -26,8 +26,8 @@

 namespace tt::umd {

-static thread_local sigjmp_buf point __attribute__((tls_model("initial-exec")));
-static thread_local volatile sig_atomic_t jump_set __attribute__((tls_model("initial-exec"))) = 0;
+static thread_local sigjmp_buf point;
+static thread_local volatile sig_atomic_t jump_set = 0;

 void sigbus_handler(int sig) {
     if (jump_set) {
