diff --git a/ttnn/cpp/ttnn/operations/creation/creation.cpp b/ttnn/cpp/ttnn/operations/creation/creation.cpp
--- a/ttnn/cpp/ttnn/operations/creation/creation.cpp
+++ b/ttnn/cpp/ttnn/operations/creation/creation.cpp
@@ -36,3 +36,3 @@ Tensor arange_impl(
     for (auto value = start; (step > 0) ? (value < stop) : (value > stop); value += step) {
-        if constexpr (std::is_same_v<T, ::bfloat16>) {
+        if constexpr (std::is_same_v<T, ::bfloat16> || std::is_same_v<T, tt::tt_metal::float16>) {
             owned_buffer[index++] = T(static_cast<float>(value));
@@ -82,2 +82,9 @@ template Tensor arange_impl<::bfloat16>(
     const MemoryConfig& output_mem_config);
+template Tensor arange_impl<tt::tt_metal::float16>(
+    const int64_t start,
+    const int64_t stop,
+    const int64_t step,
+    const Layout layout,
+    std::optional<std::reference_wrapper<MeshDevice>> device,
+    const MemoryConfig& output_mem_config);
 template Tensor arange_impl<float>(
@@ -412,2 +419,3 @@ Tensor arange(
         case DataType::BFLOAT16: return concrete_arange.template operator()<::bfloat16>();
+        case DataType::FLOAT16: return concrete_arange.template operator()<tt::tt_metal::float16>();
         case DataType::FLOAT32: return concrete_arange.template operator()<float>();
