diff --git a/tt_metal/distributed/mesh_workload.cpp b/tt_metal/distributed/mesh_workload.cpp
--- a/tt_metal/distributed/mesh_workload.cpp
+++ b/tt_metal/distributed/mesh_workload.cpp
@@ -41,3 +41,3 @@
 #include "tt_metal/distributed/fd_mesh_command_queue.hpp"
-#include "tt_metal/impl/debug/inspector/inspector.hpp"
+#include "debug/inspector/inspector.hpp"
 
diff --git a/tt_metal/impl/program/program.cpp b/tt_metal/impl/program/program.cpp
--- a/tt_metal/impl/program/program.cpp
+++ b/tt_metal/impl/program/program.cpp
@@ -70,3 +70,3 @@
 #include "tt_memory.h"
-#include "tt_metal/impl/debug/inspector/inspector.hpp"
+#include "debug/inspector/inspector.hpp"
 #include "tt_metal/impl/dispatch/data_collection.hpp"
