diff --git tt_metal/impl/tensor/host_tensor_factory.cpp tt_metal/impl/tensor/host_tensor_factory.cpp
--- tt_metal/impl/tensor/host_tensor_factory.cpp
+++ tt_metal/impl/tensor/host_tensor_factory.cpp
@@ -68,7 +68,7 @@ HostTensor host_tensor_from_span_with_pad_value(std::span<const T> buffer, Tenso
 
 template <typename T>
 HostTensor HostTensor::from_span(std::span<const T> buffer, TensorSpec spec) {
-    return host_tensor_from_span_with_pad_value(buffer, std::move(spec), T{0});
+    return host_tensor_from_span_with_pad_value(buffer, std::move(spec), T{});
 }
 
 template <typename T>
@@ -113,12 +113,12 @@ HostTensor host_tensor_from_vector_with_pad_value(std::vector<T>&& buffer, Tenso
 
 template <typename T>
 HostTensor HostTensor::from_vector(const std::vector<T>& buffer, TensorSpec spec) {
-    return host_tensor_from_vector_with_pad_value(buffer, std::move(spec), T{0});
+    return host_tensor_from_vector_with_pad_value(buffer, std::move(spec), T{});
 }
 
 template <typename T>
 HostTensor HostTensor::from_vector(std::vector<T>&& buffer, TensorSpec spec) {
-    return host_tensor_from_vector_with_pad_value(std::move(buffer), std::move(spec), T{0});
+    return host_tensor_from_vector_with_pad_value(std::move(buffer), std::move(spec), T{});
 }
 
 namespace {
diff --git tt_metal/impl/tensor/host_to_tensor_spec_apis.cpp tt_metal/impl/tensor/host_to_tensor_spec_apis.cpp
--- tt_metal/impl/tensor/host_to_tensor_spec_apis.cpp
+++ tt_metal/impl/tensor/host_to_tensor_spec_apis.cpp
@@ -171,7 +171,7 @@ HostTensor host_tensor_to_tensor_spec_with_pad_value(
 
 template <typename T>
 HostTensor to_tensor_spec(const HostTensor& tensor, const TensorSpec& dest_spec) {
-    return host_tensor_to_tensor_spec_with_pad_value(tensor, dest_spec, T{0});
+    return host_tensor_to_tensor_spec_with_pad_value(tensor, dest_spec, T{});
 }
 
 template HostTensor host_tensor_to_tensor_spec_with_pad_value<float>(const HostTensor&, const TensorSpec&, float);
