--- tt_metal/jit_build/build.cpp
+++ tt_metal/jit_build/build.cpp
@@ -493,10 +493,6 @@ void JitBuildState::compile_one(const string& out_dir, const JitBuildSettings* s
     string cmd{"cd " + out_dir + " && " + env_.gpp_};
     string defines = this->defines_;
 
-    if (env_.get_rtoptions().get_build_map_enabled()) {
-        cmd += "-save-temps=obj -fdump-tree-all -fdump-rtl-all ";
-    }
-
     if (settings) {
         // Append user args
         if (process_defines_at_compile_) {
@@ -617,7 +613,6 @@ void JitBuildState::link(const string& out_dir, const JitBuildSettings* settings
     string lflags = this->lflags_;
     if (env_.get_rtoptions().get_build_map_enabled()) {
         lflags += "-Wl,-Map=" + out_dir + this->target_name_ + ".map ";
-        lflags += "-save-temps=obj -fdump-tree-all -fdump-rtl-all ";
     }
 
     // Append user args
