--- tt_metal/jit_build/build.cpp
+++ tt_metal/jit_build/build.cpp
@@ -521,11 +521,7 @@ void JitBuildState::compile_one(const string& out_dir, const JitBuildSettings* s
     string cmd{"cd " + out_dir + " && " + env_.gpp_};
     string defines = this->defines_;
 
-    if (env_.get_rtoptions().get_build_map_enabled()) {
-        cmd += "-save-temps=obj -fdump-tree-all -fdump-rtl-all ";
-    }
-
     if (settings) {
         defines += fmt::format(R"(-DFULL_KERNEL_NAME="\"{}\"" )", settings->get_full_kernel_name());
 
         // Append user args
@@ -646,7 +642,6 @@ void JitBuildState::link(const string& out_dir, const JitBuildSettings* settings
     string lflags = this->lflags_;
     if (env_.get_rtoptions().get_build_map_enabled()) {
         lflags += "-Wl,-Map=" + out_dir + this->target_name_ + ".map ";
-        lflags += "-save-temps=obj -fdump-tree-all -fdump-rtl-all ";
     }
 
     // Append user args
