diff --git a/ttnn/cpp/ttnn/operations/data_movement/scatter/device/kernels/common.hpp b/ttnn/cpp/ttnn/operations/data_movement/scatter/device/kernels/common.hpp
--- a/ttnn/cpp/ttnn/operations/data_movement/scatter/device/kernels/common.hpp
+++ b/ttnn/cpp/ttnn/operations/data_movement/scatter/device/kernels/common.hpp
@@ -33,2 +33,7 @@ struct df_to_std<DataFormat::Float16_b> {
 
+template <>
+struct df_to_std<DataFormat::Float16> {
+    using std_type = uint16_t;
+};
+
 template <>
diff --git a/ttnn/cpp/ttnn/operations/data_movement/scatter/device/scatter_device_operation.cpp b/ttnn/cpp/ttnn/operations/data_movement/scatter/device/scatter_device_operation.cpp
--- a/ttnn/cpp/ttnn/operations/data_movement/scatter/device/scatter_device_operation.cpp
+++ b/ttnn/cpp/ttnn/operations/data_movement/scatter/device/scatter_device_operation.cpp
@@ -24,3 +24,3 @@ ScatterDeviceOperation::program_factory_t ScatterDeviceOperation::select_program
 void ScatterDeviceOperation::validate_on_program_cache_miss(
-    const operation_attributes_t& /*args*/, const tensor_args_t& tensor_args) {
+    const operation_attributes_t& args, const tensor_args_t& tensor_args) {
     const auto& input_tensor{tensor_args.input_tensor};
@@ -32,2 +32,7 @@ void ScatterDeviceOperation::validate_on_program_cache_miss(
 
+    TT_FATAL(
+        input_dtype != DataType::FLOAT16 ||
+            args.opt_reduction == ttnn::operations::data_movement::scatter::ScatterReductionType::INVALID,
+        "Float16 scatter reductions are not supported.");
+
     TT_FATAL(
