diff --git a/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_sign.h b/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_sign.h
--- a/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_sign.h
+++ b/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_sign.h
@@ -28,3 +28,4 @@ inline void calculate_sign(const uint /*exponent_size_8*/) {
         sfpi::vFloat res = 1.0f;
-        v_if(v < 0.0F) { res = -1.0f; }
+        v_if(sfpi::is_nan(v)) { res = v; }
+        v_elseif(v < 0.0F) { res = -1.0f; }
         v_elseif(_sfpu_is_fp16_zero_(v)) { res = 0.0f; }
diff --git a/tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_sign.h b/tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_sign.h
--- a/tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_sign.h
+++ b/tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_sign.h
@@ -28,3 +28,4 @@ inline void calculate_sign(const uint /*exponent_size_8*/) {
         sfpi::vFloat res = 1.0f;
-        v_if(v < 0.0F) { res = -1.0f; }
+        v_if(sfpi::is_nan(v)) { res = v; }
+        v_elseif(v < 0.0F) { res = -1.0f; }
         v_elseif(_sfpu_is_fp16_zero_(v)) { res = 0.0f; }
