diff --git a/ttnn/cpp/ttnn/operations/eltwise/unary/common/unary_op_utils.cpp b/ttnn/cpp/ttnn/operations/eltwise/unary/common/unary_op_utils.cpp
--- a/ttnn/cpp/ttnn/operations/eltwise/unary/common/unary_op_utils.cpp
+++ b/ttnn/cpp/ttnn/operations/eltwise/unary/common/unary_op_utils.cpp
@@ -789,3 +789,11 @@ std::pair<std::string, std::string> get_op_init_and_func_default(
         case UnaryOpType::ABS_INT32: return {"abs_tile_init();", fmt::format("abs_tile_int32({});", idst)};
-        case UnaryOpType::SIGN: return {"sign_tile_init();", fmt::format("sign_tile({});", idst)};
+        case UnaryOpType::SIGN:
+            if (input_dtype == DataType::INT32) {
+                return {
+                    "clamp_tile_init();", fmt::format("clamp_tile_int32({}, 0xffffffffu, 1u);", idst)};
+            }
+            if (input_dtype == DataType::UINT32) {
+                return {"nez_tile_init();", fmt::format("nez_tile_uint32({});", idst)};
+            }
+            return {"sign_tile_init();", fmt::format("sign_tile({});", idst)};
         case UnaryOpType::SQUARE:
@@ -947,3 +955,9 @@ std::map<std::string, std::string> get_defines_impl(
         {init_def, op_init_and_name.first}, {func_def, op_init_and_name.second}};
-    update_macro_defines(op_type, defines);
+    if (op_type == UnaryOpType::SIGN && input_dtype == DataType::INT32) {
+        update_macro_defines(UnaryOpType::CLAMP_TSS, defines);
+    } else if (op_type == UnaryOpType::SIGN && input_dtype == DataType::UINT32) {
+        update_macro_defines(UnaryOpType::NEZ, defines);
+    } else {
+        update_macro_defines(op_type, defines);
+    }
     return defines;
