--- tt_metal/hw/inc/internal/tt-1xx/wormhole/BUILD.bazel
+++ tt_metal/hw/inc/internal/tt-1xx/wormhole/BUILD.bazel
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+

 config_setting(
     name = "ENABLE_FWLOG",
@@ -17,7 +19,7 @@ config_setting(
 )

 headers = glob(["**/*.h", "**/*.hpp", "**/*.h"]) + ["cmds.def"]
-sources = glob(["**/*.cc", "**/*.cpp"])
+sources = glob(["**/*.cc", "**/*.cpp"], allow_empty = True)

 cc_library(
     name = "rv-wh",
