diff --git a/lib/OpModel/TTNN/TTNNOpModel.cpp b/lib/OpModel/TTNN/TTNNOpModel.cpp
index 24d9236..99e9305 100644
--- a/lib/OpModel/TTNN/TTNNOpModel.cpp
+++ b/lib/OpModel/TTNN/TTNNOpModel.cpp
@@ -511,7 +511,7 @@ auto getOpSymbol() {
   } else if constexpr (std::is_same_v<OpTy, MishOp>) {
     return WRAP_OP(::ttnn::mish);
   } else {
-    static_assert(ttmlir::utils::always_false(),
+    static_assert(ttmlir::utils::always_false<OpTy>(),
                   "add mapping from TTNN dialect to TTNN lib op");
   }
 }
