diff --git a/lib/RegisterAll.cpp b/lib/RegisterAll.cpp
index 7f03c79..51f0a72 100644
--- a/lib/RegisterAll.cpp
+++ b/lib/RegisterAll.cpp
@@ -5,24 +5,13 @@
 #include "ttmlir/RegisterAll.h"
 
 #include "ttmlir/Conversion/Passes.h"
-#include "ttmlir/Dialect/D2M/IR/D2M.h"
-#include "ttmlir/Dialect/D2M/Transforms/Passes.h"
-#include "ttmlir/Dialect/Debug/IR/Debug.h"
-#include "ttmlir/Dialect/LLVM/Transforms/Passes.h"
-#include "ttmlir/Dialect/SFPI/IR/SFPI.h"
 #include "ttmlir/Dialect/TTCore/IR/TTCore.h"
 #include "ttmlir/Dialect/TTCore/Transforms/Passes.h"
 #include "ttmlir/Dialect/TTCore/Utils/PopulateArgumentTypes.h"
 #include "ttmlir/Dialect/TTIR/IR/TTIR.h"
 #include "ttmlir/Dialect/TTIR/Pipelines/TTIRPipelines.h"
 #include "ttmlir/Dialect/TTIR/Transforms/Passes.h"
-#include "ttmlir/Dialect/TTKernel/IR/TTKernel.h"
-#include "ttmlir/Dialect/TTKernel/Pipelines/TTKernelPipelines.h"
-#include "ttmlir/Dialect/TTKernel/Transforms/Passes.h"
-// #include "ttmlir/Dialect/SFPI/Transforms/Passes.h"  // Commented out until we
-// have passes
-#include "ttmlir/Dialect/D2M/Pipelines/D2MPipelines.h"
-#include "ttmlir/Dialect/TTMetal/Transforms/Passes.h"
 #include "ttmlir/Dialect/TTNN/IR/TTNN.h"
 #include "ttmlir/Dialect/TTNN/Pipelines/TTNNPipelines.h"
 #include "ttmlir/Dialect/TTNN/Transforms/Passes.h"
@@ -86,20 +75,17 @@
 void mlir::tt::registerAllDialects(mlir::DialectRegistry &registry) {
   registry.insert<
       mlir::tt::ttcore::TTCoreDialect, mlir::tt::ttir::TTIRDialect,
-      mlir::tt::d2m::D2MDialect, mlir::tt::ttnn::TTNNDialect,
-      mlir::tt::ttmetal::TTMetalDialect, mlir::tt::ttkernel::TTKernelDialect,
-      mlir::tt::sfpi::SFPIDialect, mlir::func::FuncDialect,
-      mlir::arith::ArithDialect, mlir::math::MathDialect,
-      mlir::ml_program::MLProgramDialect, mlir::tensor::TensorDialect,
-      mlir::linalg::LinalgDialect, mlir::affine::AffineDialect,
-      mlir::scf::SCFDialect, mlir::cf::ControlFlowDialect,
-      mlir::tosa::TosaDialect, mlir::vector::VectorDialect,
-      mlir::memref::MemRefDialect, mlir::emitc::EmitCDialect,
-      mlir::bufferization::BufferizationDialect, mlir::LLVM::LLVMDialect,
-      mlir::quant::QuantDialect,
-      mlir::tt::debug::DebugDialect, mlir::pdl::PDLDialect,
+      mlir::tt::ttnn::TTNNDialect, mlir::func::FuncDialect,
+      mlir::arith::ArithDialect, mlir::math::MathDialect,
+      mlir::ml_program::MLProgramDialect, mlir::tensor::TensorDialect,
+      mlir::linalg::LinalgDialect, mlir::affine::AffineDialect,
+      mlir::scf::SCFDialect, mlir::cf::ControlFlowDialect,
+      mlir::tosa::TosaDialect, mlir::vector::VectorDialect,
+      mlir::memref::MemRefDialect, mlir::emitc::EmitCDialect,
+      mlir::bufferization::BufferizationDialect, mlir::LLVM::LLVMDialect,
+      mlir::quant::QuantDialect, mlir::pdl::PDLDialect,
       mlir::pdl_interp::PDLInterpDialect>();
 
 #if TTMLIR_ENABLE_STABLEHLO
@@ -149,25 +135,19 @@ void mlir::tt::registerAllPasses() {
 
   mlir::tt::ttcore::registerPasses();
   mlir::tt::ttcore::registerTTPopulateArgumentTypes();
   mlir::tt::ttir::registerPasses();
-  mlir::tt::d2m::registerPasses();
   mlir::tt::ttnn::registerTTNNOptimizer();
   mlir::tt::ttnn::registerPasses();
-  mlir::tt::ttmetal::registerPasses();
-  mlir::tt::ttkernel::registerPasses();
-  mlir::tt::llvm_util::registerPasses();
   mlir::tt::transforms::registerPasses();
 
 #if TTMLIR_ENABLE_STABLEHLO
   mlir::tt::stablehlo::registerPasses();
   mlir::stablehlo::registerStablehloComplexMathExpanderPass();
 #endif
 
   // Register pipelines.
   mlir::tt::ttir::registerTTIRPipelines();
   mlir::tt::ttnn::registerTTNNPipelines();
-  mlir::tt::ttmetal::registerD2MPipelines();
-  mlir::tt::ttkernel::registerTTKernelPipelines();
 
 #if TTMLIR_ENABLE_STABLEHLO
   // Register all SDY passes and pipelines.
