diff --git a/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp b/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp
--- a/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp
+++ b/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp
@@ -689,6 +689,2 @@ TTNNOperandsWorkaroundsFactory::createTanhOpOperandsWorkarounds() {
   TTNNOperandWorkarounds operandWorkaround;
-  // Tanh op accurate mode requires bfloat16 data type.
-  // Issue: https://github.com/tenstorrent/tt-metal/issues/22593
-  operandWorkaround.tensorDataTypeWorkaround =
-      mlir::tt::ttcore::DataType::BFloat16;
 
diff --git a/runtime/lib/ttnn/operations/eltwise/unary/unary.cpp b/runtime/lib/ttnn/operations/eltwise/unary/unary.cpp
--- a/runtime/lib/ttnn/operations/eltwise/unary/unary.cpp
+++ b/runtime/lib/ttnn/operations/eltwise/unary/unary.cpp
@@ -159,5 +159,9 @@ void run(const ::tt::target::ttnn::EltwiseUnaryOp *op,
   case ::tt::target::ttnn::EltwiseUnaryOpType::Gelu: {
-    runEltwiseUnaryWithFastAndApproximateModeOp(
-        op, tensorPool, [](auto &&...args) {
-          return ::ttnn::gelu(std::forward<decltype(args)>(args)...);
+    runEltwiseUnaryOp(
+        op, tensorPool,
+        [](const auto &input, const auto &memoryConfig, const auto &output,
+           const auto &subCoreGrids) {
+          return ::ttnn::gelu(
+              input, ::ttnn::operations::unary::GeluVariant::TANH,
+              memoryConfig, output, subCoreGrids);
         });
