diff --git a/runtime/lib/ttnn/operations/trace/capture_or_execute_trace.cpp b/runtime/lib/ttnn/operations/trace/capture_or_execute_trace.cpp
index 89f00f1..d1e55dc 100644
--- a/runtime/lib/ttnn/operations/trace/capture_or_execute_trace.cpp
+++ b/runtime/lib/ttnn/operations/trace/capture_or_execute_trace.cpp
@@ -12,5 +12,6 @@
 #include "tt/runtime/workarounds.h"
 #include "ttnn/tensor/tensor_impl.hpp"
+#include "ttnn/operations/data_movement/copy/copy.hpp"

 namespace tt::runtime::ttnn::operations::trace {

@@ -39,8 +40,7 @@ static void copyTensorFromDeviceToDevice(const ::ttnn::Tensor &srcTensor,
   LOG_ASSERT(srcTensor.storage_type() == ::ttnn::StorageType::DEVICE &&
                  dstTensor.storage_type() == ::ttnn::StorageType::DEVICE,
              "srcTensor must be on device and dstTensor must be on device");
-  ::ttnn::Tensor hostSrcTensor = ::ttnn::from_device(srcTensor);
-  ::tt::tt_metal::copy_to_device(hostSrcTensor, dstTensor);
+  ::ttnn::copy(srcTensor, dstTensor);
 }

 static void runTraceProgramAndCaptureTrace(
