diff --git a/include/ttmlir/Conversion/TTNNToEmitC/EmitCConversion.h b/include/ttmlir/Conversion/TTNNToEmitC/EmitCConversion.h
--- a/include/ttmlir/Conversion/TTNNToEmitC/EmitCConversion.h
+++ b/include/ttmlir/Conversion/TTNNToEmitC/EmitCConversion.h
@@ -2187,6 +2187,6 @@ public:
   template <
       typename MLIRAttrTy, typename = std::void_t<TTNNTargetT<MLIRAttrTy>>,
       typename =
-          std::enable_if_t<std::is_convertible_v<MLIRAttrTy, mlir::Attribute>>>
+          std::enable_if_t<std::is_convertible_v<MLIRAttrTy &, mlir::Attribute>>>
   mlir::Attribute emit(MLIRAttrTy attr) {
     auto convertedValue =
