diff --git a/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp b/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp
--- a/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp
+++ b/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp
@@ -929,7 +929,7 @@ class D2MNamedElementwiseRewriter final
       D2MNamedRewriterCommon {
 
 public:
-  D2MNamedElementwiseRewriter<ConcreteOp, TileOp>(
+  D2MNamedElementwiseRewriter(
       const TypeConverter &typeConverter, mlir::MLIRContext *ctx,
       ttcore::MemorySpace defaultInputMemSpace,
       ttcore::MemorySpace defaultOutputMemSpace, bool ttnnMode,
@@ -2091,7 +2091,7 @@ class D2MNamedTileReduceRewriter final
       D2MNamedRewriterCommon {
 
 public:
-  D2MNamedTileReduceRewriter<ConcreteOp, FloatTileOp, IntTileOp>(
+  D2MNamedTileReduceRewriter(
       const TypeConverter &typeConverter, mlir::MLIRContext *ctx,
       ttcore::MemorySpace defaultInputMemSpace,
       ttcore::MemorySpace defaultOutputMemSpace, bool ttnnMode,
