diff --git a/include/ttmlir/Dialect/StableHLO/Utils/ShardingUtils.h b/include/ttmlir/Dialect/StableHLO/Utils/ShardingUtils.h
--- a/include/ttmlir/Dialect/StableHLO/Utils/ShardingUtils.h
+++ b/include/ttmlir/Dialect/StableHLO/Utils/ShardingUtils.h
@@ -31,13 +31,16 @@ inline const llvm::SmallVector<llvm::SmallVector<int64_t, 2>> SupportedMeshes =
       {1, 16},
       {1, 32},
       {1, 64},
+      {2, 1},
       {2, 2},
       {2, 4},
       {2, 8},
+      {4, 1},
       {4, 4},
       {4, 8},
       {4, 16},
       {4, 32},
+      {8, 1},
       {8, 4},
       {8, 8},
       {8, 16}}};
