diff --git a/runtime/lib/ttnn/operations/eltwise/unary/unary.cpp b/runtime/lib/ttnn/operations/eltwise/unary/unary.cpp
index 8378104..32eca62 100644
--- a/runtime/lib/ttnn/operations/eltwise/unary/unary.cpp
+++ b/runtime/lib/ttnn/operations/eltwise/unary/unary.cpp
@@ -157,9 +157,13 @@ void run(const ::tt::target::ttnn::EltwiseUnaryOp *op,
     break;
   }
   case ::tt::target::ttnn::EltwiseUnaryOpType::Gelu: {
-    runEltwiseUnaryWithFastAndApproximateModeOp(
-        op, tensorPool, [](auto &&...args) {
-          return ::ttnn::gelu(std::forward<decltype(args)>(args)...);
+    runEltwiseUnaryOp(
+        op, tensorPool,
+        [](const auto &input, const auto &memoryConfig, const auto &output,
+           const auto &subCoreGrids) {
+          return ::ttnn::gelu(
+              input, ::ttnn::operations::unary::GeluVariant::TANH,
+              memoryConfig, output, subCoreGrids);
         });
     break;
   }
diff --git a/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp b/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp
index c1fb81d..f0e33a7 100644
--- a/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp
+++ b/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp
@@ -687,10 +687,6 @@ TTNNOperandsWorkaroundsFactory::createRotaryEmbeddingOpOperandsWorkarounds(
 TTNNOperandsWorkarounds
 TTNNOperandsWorkaroundsFactory::createTanhOpOperandsWorkarounds() {
   TTNNOperandWorkarounds operandWorkaround;
-  // Tanh op accurate mode requires bfloat16 data type.
-  // Issue: https://github.com/tenstorrent/tt-metal/issues/22593
-  operandWorkaround.tensorDataTypeWorkaround =
-      mlir::tt::ttcore::DataType::BFloat16;
 
   return TTNNOperandsWorkarounds::createEmptyTTNNOperandsWorkarounds()
       .addInputOperandWorkaround(operandWorkaround)
