diff --git a/lib/RegisterAll.cpp b/lib/RegisterAll.cpp
--- a/lib/RegisterAll.cpp
+++ b/lib/RegisterAll.cpp
@@ -10,3 +10,2 @@
 #include "ttmlir/Dialect/Debug/IR/Debug.h"
-#include "ttmlir/Dialect/EmitPy/IR/EmitPy.h"
 #include "ttmlir/Dialect/LLVM/Transforms/Passes.h"
@@ -99,3 +98,3 @@ void mlir::tt::registerAllDialects(mlir::DialectRegistry &registry) {
       mlir::bufferization::BufferizationDialect, mlir::LLVM::LLVMDialect,
-      mlir::quant::QuantDialect, mlir::tt::emitpy::EmitPyDialect,
+      mlir::quant::QuantDialect,
       mlir::tt::debug::DebugDialect, mlir::pdl::PDLDialect,
@@ -145,5 +144,2 @@ void mlir::tt::registerAllExtensions(mlir::DialectRegistry &registry) {
 void mlir::tt::registerAllPasses() {
-  // Register all dialect conversion passes.
-  mlir::tt::registerTTMLIRConversionPasses();
-
   // Registering -remove-dead-values built-in mlir pass to optimize out the
