diff --git a/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp b/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp
index 7cc6d63..3054ea6 100644
--- a/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp
+++ b/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp
@@ -916,7 +916,7 @@ class D2MNamedElementwiseRewriter final
       D2MNamedRewriterCommon {
 
 public:
-  D2MNamedElementwiseRewriter<ConcreteOp, TileOp>(
+  D2MNamedElementwiseRewriter(
       const TypeConverter &typeConverter, mlir::MLIRContext *ctx,
       ttcore::MemorySpace defaultInputMemSpace,
       ttcore::MemorySpace defaultOutputMemSpace, bool ttnnMode,
@@ -2078,7 +2078,7 @@ class D2MNamedTileReduceRewriter final
       D2MNamedRewriterCommon {
 
 public:
-  D2MNamedTileReduceRewriter<ConcreteOp, FloatTileOp, IntTileOp>(
+  D2MNamedTileReduceRewriter(
       const TypeConverter &typeConverter, mlir::MLIRContext *ctx,
       ttcore::MemorySpace defaultInputMemSpace,
       ttcore::MemorySpace defaultOutputMemSpace, bool ttnnMode,
