diff --git a/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp b/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp
index 8fbd9aa4..7b96b2a2 100644
--- a/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp
+++ b/lib/Dialect/TTNN/IR/TTNNWorkaroundsPass.cpp
@@ -752,24 +752,6 @@ TTNNOperandsWorkaroundsFactory::createGroupNormOpOperandsWorkarounds(
   return workarounds;
 }
 
-// Helper function to determine if data type workaround is required for a unary
-// bitwise op. Set the workaround data type based on the unary bitwise op.
-static std::optional<mlir::tt::ttcore::DataType>
-unaryBitwiseOpDTypeWorkaround(mlir::Operation *op, mlir::Type elementType) {
-  mlir::tt::ttcore::DataType dType =
-      mlir::tt::ttcore::elementTypeToDataType(elementType);
-
-  // tt-metal issue: https://github.com/tenstorrent/tt-metal/issues/31373
-  if (isa<ttnn::BitwiseNotOp>(op)) {
-    if (dType == mlir::tt::ttcore::DataType::UInt8 ||
-        dType == mlir::tt::ttcore::DataType::UInt16 ||
-        dType == mlir::tt::ttcore::DataType::UInt32) {
-      return mlir::tt::ttcore::DataType::Int32;
-    }
-  }
-  return {};
-}
-
 // Factory method to create a set of workarounds for unary bitwise operation
 // (BitwiseNotOp) operands.
 TTNNOperandsWorkarounds
@@ -780,10 +762,6 @@ TTNNOperandsWorkaroundsFactory::createUnaryBitwiseOpOperandsWorkarounds(
       mlir::cast<mlir::RankedTensorType>(op->getOperand(0).getType());
 
   TTNNOperandWorkarounds operandWorkaround;
-  if (auto dtype =
-          unaryBitwiseOpDTypeWorkaround(op, inputType.getElementType())) {
-    operandWorkaround.tensorDataTypeWorkaround = *dtype;
-  }
   if (!mlir::cast<ttnn::TTNNLayoutAttr>(inputType.getEncoding()).isTiled()) {
     operandWorkaround.tensorLayoutWorkaround = Layout::Tile;
   }
