diff --git a/lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp b/lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp
--- a/lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp
+++ b/lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp
@@ -302,5 +302,10 @@ private:
                           mlir::stablehlo::ReduceOp::Adaptor adaptor,
                           ConversionPatternRewriter &rewriter) const {
+    if (srcOp.getInputs().size() != 1) {
+      return rewriter.notifyMatchFailure(
+          srcOp, "variadic stablehlo.reduce is not supported");
+    }
+
     if (!srcOp.getBody().hasOneBlock()) {
       return rewriter.notifyMatchFailure(
           srcOp,
