diff --git a/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp b/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp
--- a/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp
+++ b/lib/Conversion/TTIRToD2M/TTIRToD2M.cpp
@@ -931,3 +931,3 @@ class D2MNamedElementwiseRewriter final
 public:
-  D2MNamedElementwiseRewriter<ConcreteOp, TileOp>(
+  D2MNamedElementwiseRewriter(
       const TypeConverter &typeConverter, mlir::MLIRContext *ctx,
@@ -2093,3 +2093,3 @@ class D2MNamedTileReduceRewriter final
 public:
-  D2MNamedTileReduceRewriter<ConcreteOp, FloatTileOp, IntTileOp>(
+  D2MNamedTileReduceRewriter(
       const TypeConverter &typeConverter, mlir::MLIRContext *ctx,
