diff --git a/pjrt_implementation/inc/api/buffer_instance.h b/pjrt_implementation/inc/api/buffer_instance.h
--- a/pjrt_implementation/inc/api/buffer_instance.h
+++ b/pjrt_implementation/inc/api/buffer_instance.h
@@ -324,2 +324,5 @@ PJRT_Error *onBufferMemory(PJRT_Buffer_Memory_Args *args);
 
+// Implements PJRT_Buffer_UnsafePointer API function.
+PJRT_Error *onBufferUnsafePointer(PJRT_Buffer_UnsafePointer_Args *args);
+
 // Implements PJRT_Buffer_ReadyEvent API function.
diff --git a/pjrt_implementation/src/api/buffer_instance.cc b/pjrt_implementation/src/api/buffer_instance.cc
--- a/pjrt_implementation/src/api/buffer_instance.cc
+++ b/pjrt_implementation/src/api/buffer_instance.cc
@@ -115,2 +115,3 @@ void BufferInstance::bindApi(PJRT_Api *api) {
   api->PJRT_Buffer_ReadyEvent = internal::onBufferReadyEvent;
+  api->PJRT_Buffer_UnsafePointer = internal::onBufferUnsafePointer;
   api->PJRT_Buffer_Memory = internal::onBufferMemory;
@@ -725,2 +726,17 @@ PJRT_Error *onBufferDevice(PJRT_Buffer_Device_Args *args) {
 
+PJRT_Error *onBufferUnsafePointer(PJRT_Buffer_UnsafePointer_Args *args) {
+  ZoneScoped;
+  DLOG_F(LOG_DEBUG, "BufferInstance::PJRT_Buffer_UnsafePointer");
+
+  BufferInstance *buffer = BufferInstance::unwrap(args->buffer);
+  const PjrtTensor *tensor = buffer->getPjrtTensor().get();
+  if (tensor == nullptr || buffer->isDataDeleted()) {
+    return *ErrorInstance::makeError(tt_pjrt_status::kInvalidArgument)
+                .release();
+  }
+
+  args->buffer_pointer = reinterpret_cast<uintptr_t>(tensor);
+  return nullptr;
+}
+
 PJRT_Error *onBufferMemory(PJRT_Buffer_Memory_Args *args) {
