diff --git a/pjrt_implementation/inc/api/buffer_instance.h b/pjrt_implementation/inc/api/buffer_instance.h
--- a/pjrt_implementation/inc/api/buffer_instance.h
+++ b/pjrt_implementation/inc/api/buffer_instance.h
@@ -300,2 +300,6 @@ onBufferDynamicDimensionIndices(PJRT_Buffer_DynamicDimensionIndices_Args *args);
 
+// Implements PJRT_Buffer_OnDeviceSizeInBytes API function.
+PJRT_Error *
+onBufferOnDeviceSizeInBytes(PJRT_Buffer_OnDeviceSizeInBytes_Args *args);
+
 // Implements PJRT_Buffer_ToHostBuffer API function.
diff --git a/pjrt_implementation/src/api/buffer_instance.cc b/pjrt_implementation/src/api/buffer_instance.cc
--- a/pjrt_implementation/src/api/buffer_instance.cc
+++ b/pjrt_implementation/src/api/buffer_instance.cc
@@ -107,2 +107,4 @@ void BufferInstance::bindApi(PJRT_Api *api) {
       internal::onBufferDynamicDimensionIndices;
+  api->PJRT_Buffer_OnDeviceSizeInBytes =
+      internal::onBufferOnDeviceSizeInBytes;
   api->PJRT_Buffer_ToHostBuffer = internal::onBufferToHostBuffer;
@@ -605,2 +607,13 @@ PJRT_Error *onBufferDynamicDimensionIndices(
 
+PJRT_Error *
+onBufferOnDeviceSizeInBytes(PJRT_Buffer_OnDeviceSizeInBytes_Args *args) {
+  ZoneScoped;
+  DLOG_F(LOG_DEBUG, "BufferInstance::PJRT_Buffer_OnDeviceSizeInBytes");
+
+  const BufferInstance *buffer = BufferInstance::unwrap(args->buffer);
+  args->on_device_size_in_bytes = buffer->logicalTensorSize();
+
+  return nullptr;
+}
+
 PJRT_Error *onBufferToHostBuffer(PJRT_Buffer_ToHostBuffer_Args *args) {
