--- a/pjrt_implementation/src/api/module_builder/module_builder.cc
+++ b/pjrt_implementation/src/api/module_builder/module_builder.cc
@@ -934,2 +934,3 @@ tt_pjrt_status ModuleBuilder::convertFromTTIRToTTNN(
     options.experimentalWeightDtype = mlir::tt::ttnn::BFPDtype::BFP_BFloat8;
+    options.enableActivationDtypeLowering = true;
   } else if (compile_options.experimental_weight_dtype == "bfp_bf4") {
