diff --git a/pjrt_implementation/src/api/buffer_instance.cc b/pjrt_implementation/src/api/buffer_instance.cc
--- a/pjrt_implementation/src/api/buffer_instance.cc
+++ b/pjrt_implementation/src/api/buffer_instance.cc
@@ -397,9 +397,4 @@ BufferInstance::calculateShape(const std::int64_t *dims, size_t num_dims,
                                PJRT_Buffer_Type data_type) {
-  if (data_type_utils::isComplexPJRTType(data_type) && num_dims == 0) {
-    // Throw error if complex tensor num_dims == 0.
-    TT_THROW("Complex tensor with num_dims == 0 is not supported.");
-  }
-
   std::vector<std::uint32_t> shape;
   for (size_t i = 0; i < num_dims; ++i) {
     shape.push_back(dims[i]);
